src/IB/IBFESurfaceMethod.cpp

```diff
@@ -557,32 +765,76 @@ IBFESurfaceMethod::interpolateVelocity(const int u_data_idx,
+        boost::array<System*, NDIM> DU_j_system;
+        boost::array<const DofMap*, NDIM> DU_j_dof_map;
+        boost::array<FEDataManager::SystemDofMapCache*, NDIM> DU_j_dof_map_cache;
+        boost::array<std::vector<std::vector<unsigned int> >, NDIM> DU_j_dof_indices;
+        FEType DU_j_fe_type; 
+        std::vector<std::vector<unsigned int> > WSS_o_dof_indices(NDIM);
+        System* WSS_o_system;
+        const DofMap* WSS_o_dof_map;
+        FEDataManager::SystemDofMapCache* WSS_o_dof_map_cache;
+        std::vector<std::vector<unsigned int> > WSS_i_dof_indices(NDIM);
+        System* WSS_i_system;
+        const DofMap* WSS_i_dof_map;
+        FEDataManager::SystemDofMapCache* WSS_i_dof_map_cache;
+
+        if (d_use_velocity_jump_conditions)
+        {
+            for (unsigned int i = 0; i < NDIM; ++i)
+            {
+                DU_j_system[i] = &equation_systems->get_system(VELOCITY_JUMP_SYSTEM_NAME[i]);
+                DU_j_dof_map[i] = &DU_j_system[i]->get_dof_map();
+                DU_j_dof_map_cache[i] = d_fe_data_managers[part]->getDofMapCache(VELOCITY_JUMP_SYSTEM_NAME[i]);
+                DU_j_fe_type = DU_j_dof_map[i]->variable_type(0);
+                for (unsigned int d = 0; d < NDIM; ++d)
+                {
+                    TBOX_ASSERT(DU_j_dof_map[i]->variable_type(d) == DU_j_fe_type);
+                }
+               // TBOX_ASSERT(U_fe_type == DU_j_fe_type);
+                DU_j_dof_indices[i].resize(NDIM);
+            }
+
+            WSS_o_system = &equation_systems->get_system(WSS_O_SYSTEM_NAME);
+            WSS_o_dof_map = &WSS_o_system->get_dof_map();
+            WSS_o_dof_map_cache = d_fe_data_managers[part]->getDofMapCache(WSS_O_SYSTEM_NAME);
+            FEType WSS_o_fe_type = WSS_o_dof_map->variable_type(0);
+            for (unsigned int d = 0; d < NDIM; ++d)
+            {
+                TBOX_ASSERT(WSS_o_dof_map->variable_type(d) == WSS_o_fe_type);
+            }
+
+            WSS_i_system = &equation_systems->get_system(WSS_I_SYSTEM_NAME);
+            WSS_i_dof_map = &WSS_i_system->get_dof_map();
+            WSS_i_dof_map_cache = d_fe_data_managers[part]->getDofMapCache(WSS_I_SYSTEM_NAME);
+            FEType WSS_i_fe_type = WSS_i_dof_map->variable_type(0);
+            for (unsigned int d = 0; d < NDIM; ++d)
+            {
+                TBOX_ASSERT(WSS_i_dof_map->variable_type(d) == WSS_i_fe_type);
+            }
+        }
         FEType fe_type = U_fe_type;
         UniquePtr<FEBase> fe = FEBase::build(dim, fe_type);
         const std::vector<double>& JxW = fe->get_JxW();
         const std::vector<std::vector<double> >& phi = fe->get_phi();
         boost::array<const std::vector<std::vector<double> >*, NDIM - 1> dphi_dxi;
         dphi_dxi[0] = &fe->get_dphidxi();
         if (NDIM > 2) dphi_dxi[1] = &fe->get_dphideta();
+        
+        
+        FEType fe2_type = DU_j_fe_type;
+        UniquePtr<FEBase> fe2 = FEBase::build(dim, fe2_type);
```

Comment:
Also cosmetic: we should use `std::unique_ptr` now.

Reply:
There's a perl script in the scripts directory that will replace UniquePtr with unique_ptr.

Reply:
```suggestion
        std::unique_ptr<FEBase> fe_DU_jump = FEBase::build(dim, DU_jump_fe_type);
```